rocAL/source/readers/file_source_reader.cpp

```diff
@@ -163,37 +208,43 @@ Reader::Status FileSourceReader::generate_file_names() {
 
     auto ret = Reader::Status::OK;
     if (!_file_list_path.empty()) {  // Reads the file paths from the file list and adds to file_names vector for decoding
-        std::ifstream fp(_file_list_path);
-        if (fp.is_open()) {
-            while (fp) {
-                std::string file_label_path;
-                std::getline(fp, file_label_path);
-                std::istringstream ss(file_label_path);
-                std::string file_path;
-                std::getline(ss, file_path, ' ');
+        if (_meta_data_reader) {
+            auto vec_rel_file_path = _meta_data_reader->get_relative_file_path();  // Get the relative file path's from meta_data_reader
+            for (auto file_path : vec_rel_file_path) {
                 if (filesys::path(file_path).is_relative()) {  // Only add root path if the file list contains relative file paths
                     if (!filesys::exists(_folder_path))
                         THROW("File list contains relative paths but root path doesn't exists");
-                    file_path = _folder_path + "/" + file_path;
+                    _absolute_file_path = _folder_path + "/" + file_path;
                 }
-                std::string file_name = file_path.substr(file_path.find_last_of("/\\") + 1);
+                if (filesys::is_regular_file(_absolute_file_path)) {
+                    _last_file_name = _absolute_file_path;
```

Comment:
Is this executed for every file in the file system? We need to set it only after finishing all the files.